code/modules/transport/pods/secondary_system.dm

```diff
@@ -585,6 +594,24 @@
 			src.ship.myhud.update_states()
 		boutput(usr, SPAN_NOTICE("Thrusters will now provide ship movement to the [src.turn_dir]."))
 
+/obj/item/shipcomponent/secondary_system/thrusters/afterburner
+	name = "Afterburner"
+	desc = "An engine augment that enhances the burning of plasma, increasing speed for a short duration."
+	hud_state = "lat_thrusters_right"
+	f_active = TRUE
+	power_used = 50
+
+	use_thrusters(mob/user)
+		if (ON_COOLDOWN(src, "thruster_movement", 30 SECONDS))
+			boutput(user, "[src.ship.ship_message("Afterburner is recharging! [round(GET_COOLDOWN(src, "thruster_movement") / 10, 0.1)] seconds left.")]")
+			return
+
+		// spawn to allow button clunk sound to play right away
+		SPAWN(0)
+			for (var/i in 1 to 10)
+				step(src.ship, src.ship.dir)
+				sleep(1 SECOND)
+
```

Comment:
This feels pretty kludgy - would rather see it interact with the movement controller (temporarily raise accel/max speed? Force additional velocity in facing direction?) vs a step-sleep loop

Reply:
Sure, will revise